components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,340 @@
+
+            ErrorMessages error = ErrorMessages.ERROR_WHILE_ADDING_AUTHENTICATOR;
+            throw new AuthenticatorMgtServerException(error.getMessage(), error.getMessage(), error.getCode(), e);
+        } finally {
+            IdentityDatabaseUtil.closeConnection(dbConnection);
+        }
+    }
+
+    @Override
+    public LocalAuthenticatorConfig updateUserDefinedLocalAuthenticator(LocalAuthenticatorConfig
+                existingAuthenticatorConfig, LocalAuthenticatorConfig updatedAuthenticatorConfig, Integer tenantId)
+                throws AuthenticatorMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+        try {
+
+            if (isBasicInfoUpdated(existingAuthenticatorConfig, updatedAuthenticatorConfig)) {
+                try (NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection,
+                        Query.UPDATE_AUTHENTICATOR_SQL)) {
+
+                    statement.setString(Column.DISPLAY_NAME, updatedAuthenticatorConfig.getDisplayName());
+                    int isEnabled = updatedAuthenticatorConfig.isEnabled() ? 1 : 0;
+                    statement.setInt(Column.IS_ENABLED, isEnabled);
+                    statement.setString(Column.NAME, existingAuthenticatorConfig.getName());
+                    statement.setInt(Column.TENANT_ID, tenantId);
+                    statement.executeUpdate();
+                }
+            }
+
+            // Will delete all the properties of given authenticator and add the updated properties.
+            int configId = getAuthenticatorIdentifier(dbConnection, existingAuthenticatorConfig.getName(), tenantId);
+            try (NamedPreparedStatement statementDeleteProp = new NamedPreparedStatement(dbConnection,
+                    Query.DELETE_AUTHENTICATOR_PROP_SQL)) {
+
+                statementDeleteProp.setInt(Column.AUTHENTICATOR_ID, configId);
+                statementDeleteProp.setInt(Column.TENANT_ID, tenantId);
+                statementDeleteProp.executeUpdate();
+            }
+
+            try (NamedPreparedStatement statementProp = new NamedPreparedStatement(dbConnection,
+                    Query.ADD_AUTHENTICATOR_PROP_SQL)) {
+
+                for (Property prop : updatedAuthenticatorConfig.getProperties()) {
+                    statementProp.setInt(Column.AUTHENTICATOR_ID, configId);
+                    statementProp.setInt(Column.TENANT_ID, tenantId);
+                    statementProp.setString(Column.PROPERTY_KEY, prop.getName());
+                    statementProp.setString(Column.PROPERTY_VALUE, prop.getValue());
+                    if (prop.isConfidential()) {
+                        statementProp.setString(Column.IS_SECRET, IS_TRUE_VALUE);
+                    } else {
+                        statementProp.setString(Column.IS_SECRET, IS_FALSE_VALUE);
+                    }
+                    statementProp.executeUpdate();
+                }
+            }
```

Comment:
This also can be extracted for readability

Reply:
Addressed with commit d72ebd31b7150e7c562faf83ee80300a60e1e5d8